### DISCUSSION TITLE
use `sqlx.types.Json<T>` as a schema in utoipa

### DISCUSSION THREAD

**Discussion by vncsalencar** [juhaku/utoipa#650] [Q&A]
When using a struct that has `sqlx.types.Json<T>`, utoipa throws `
`Could not resolve reference: Could not resolve pointer: /components/schemas/sqlx.types.Json does not exist in document`.
How does one deal with it?

**Comment by jayvdb** [comment]
Three solution: 
1. add sqlx support to utoipa
2. add utoipa support to sqlx
3. use utoipa's `#[schema(schema_with = ...)]` on your struct field to glue in support for sqlx fields.

e.g. 
```rust
#[derive(..., utoipa::ToSchema)]
struct Foo {
    ...
    #[schema(schema_with = create_sqlx_json_schema::<Bar>)]
    pub bar: sqlx.types.Json<Bar>,
    ...
}

pub fn create_sqlx_json_schema<T: for<'a> ToSchema<'a>>() -> Object {
    ObjectBuilder::new()
        .property("inner", ...)
        .required("inner")
        .build()
}
```
